images/disks-images-provider/entrypoint.sh

```diff
@@ -28,7 +28,16 @@ mkdir -p /images/datavolume1 /images/datavolume2 /images/datavolume3
 echo "converting cirros image from qcow2 to raw, and copying it to local-storage directory, and creating a loopback device from it"
 # /local-storage will be mapped to the host dir, which will also be used by the local storage provider
 qemu-img convert -f qcow2 -O raw /images/cirros/disk.img /local-storage/cirros.img.raw
+
+# When start the pod in kind env, as /dev folder is shared among all containers, there is no necessary to create multi loop device for one image
+# Check if cirros.image.raw already setup loop device, if not create one, otherwise use existed one
+DEVICE=$(losetup -l| { grep /local-storage/cirros.img.raw || true; } |awk '{print $1}'|sed -n '1p')
+echo ${DEVICE}
+if [ "${DEVICE}" != "" ];then
+LOOP_DEVICE="${DEVICE}"
```

Comment:
You are suggesting to use the same loop device on both nodes?
What if at some point one will ask to edit the mounted image?

Also, the error is due to the leaking loop devices. If you run the cluster, deploy kubevirt (make cluster-sync) and take it down, you will notice the leak (losetup -l). Once it reaches 100, the error will start appearing.

Reply:
> You are suggesting to use the same loop device on both nodes?
> What if at some point one will ask to edit the mounted image?
> 
> Also, the error is due to the leaking loop devices. If you run the cluster, deploy kubevirt (make cluster-sync) and take it down, you will notice the leak (losetup -l). Once it reaches 100, the error will start appearing.

Hi, @EdDev, yes, you are right！ I just verify it, the root cause for the error is loopdevice leaking. I have stuck here for a long time. 
When I did functest in kind-k8s-1.17 kubevirt cluster, the disks-images-providers pods always went failure. "/cirros.img.raw " would bind all loop devices. Do you have any idea where the leak comes from?
Another question is why creating more than 100 loop device in container lead to this error. I checked the loopdevice limit in host, I can create more than 300 loopdevice in host. 

Looking forward your reply and sorry about the mistake.